xmodule/capa/xqueue_submission.py

```diff
@@ -0,0 +1,104 @@
+"""
+This module provides an interface for submitting student responses
+to an external grading system through XQueue.
+"""
+
+import json
+import logging
+
+from xmodule.capa.errors import (
+    GetSubmissionParamsError,
+    JSONParsingError,
+    MissingKeyError,
+    ValidationError,
+    TypeErrorSubmission,
+    RuntimeErrorSubmission
+)
+
+log = logging.getLogger(__name__)
+
+
+class XQueueInterfaceSubmission:
+    """Interface to the external grading system."""
+
+    def __init__(self, block):
+        self.block = block
+
+    def _parse_json(self, data, name):
+        """Helper function to parse JSON safely."""
+        try:
+            return json.loads(data) if isinstance(data, str) else data
```

Comment:
Please explain in a docstring why this is necessary, i.e. why do we not know whether we should be parsing something as JSON or sending it back as-is?